34.md

```diff
@@ -103,9 +108,64 @@ The first patch revision in a patch revision SHOULD include a [NIP-10](10.md) `e
 
 The first patch in a series MAY be a cover letter in the format produced by `git format-patch`.
 
+### Pull Requests
+
+The PR or PR update tip SHOULD be successfully pushed to all repositories listed in its `clone` tag before the event is signed. Clients SHOULD fetch by commit-id rather than ref name.
+
+An attempt SHOULD be made to push tip to `refs/nostr/<[PR|PR-Update]-event-id>` for each [grasp server](https://njump.me/naddr1qvzqqqrhnypzpgqgmmc409hm4xsdd74sf68a2uyf9pwel4g9mfdg8l5244t6x4jdqy28wumn8ghj7un9d3shjtnwva5hgtnyv4mqqpt8wfshxuqlnvh8x) identified in the repository's announcement event's `"clone"` tag using identification method: `clone` tag includes `[http|https]://<grasp-path>/<valid-npub>/<string>.git` and `relays` tag includes `[ws/wss]://<grasp-path>`.
+
+Clients MAY fallback to creating a 'personal-fork' `repository announcement` listing other grasp servers, e.g. from the `User grasp list`, for the purpose of serving the specified commit(s).
+
+```jsonc
+{
+  "kind": 1618,
+  "content": "<markdown text>",
+  "tags": [
+    ["a", "30617:<base-repo-owner-pubkey>:<base-repo-id>"],
+    ["r", "<earliest-unique-commit-id-of-repo>"] // so clients can subscribe to all PRs sent to a local git repo
+    ["p", "<repository-owner>"],
+    ["p", "<other-user>"], // optionally send the PR to another user to bring it to their attention
+
+    ["subject", "<PR-subject>"]
+    ["t", "<PR-label>"] // optional
+    ["t", "<another-PR-label>"] // optional
+
+    ["c", "<current-commit-id>"] // tip of the PR branch
+    ["clone", "<clone-url>", ...] // at least one git clone url where commit can be downloaded
+    ["branch-name", "<branch-name>"] // optional recommended branch name
+
+    ["e", "<root-patch-event-id>"] // optionally indicate PR is a revision of an existing patch, which should be closed
```

Comment:
So, when I want to fetch a patch, do I have to check if it's mentioned in a PR to fetch the PR's status? Isn't that complicated? I mean, there's no need to do all of this; we can just close the patch and open a PR. There's no need to check if the PR exists and fetch its status in every patch, right?

Also if there is more than one PR mentioning the same patch, which PR should I fetch its status? It's really complicated.

Knowing that opening a patch and replacing it with a PR later is rare, because I'll know the changes I made before submitting the patch. I don't think there are reviewers who would request changes so numerous that they would require moving the patch to a PR.

I know I've annoyed you with this point, but it really annoys me. Sorry.